docs/book/src/cronjob-tutorial/testdata/project/internal/controller/cronjob_controller.go

```diff
@@ -325,78 +251,15 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	*/
 
-	/*
-		We'll calculate the next scheduled time using our helpful cron library.
-		We'll start calculating appropriate times from our last run, or the creation
-		of the CronJob if we can't find a last run.
-
-		If there are too many missed runs and we don't have any deadlines set, we'll
-		bail so that we don't cause issues on controller restarts or wedges.
-
-		Otherwise, we'll just return the missed runs (of which we'll just use the latest),
-		and the next run, so that we can know when it's time to reconcile again.
-	*/
-	getNextSchedule := func(cronJob *batchv1.CronJob, now time.Time) (lastMissed time.Time, next time.Time, err error) {
-		sched, err := cron.ParseStandard(cronJob.Spec.Schedule)
-		if err != nil {
-			return time.Time{}, time.Time{}, fmt.Errorf("Unparseable schedule %q: %v", cronJob.Spec.Schedule, err)
-		}
-
-		// for optimization purposes, cheat a bit and start from our last observed run time
-		// we could reconstitute this here, but there's not much point, since we've
-		// just updated it.
-		var earliestTime time.Time
-		if cronJob.Status.LastScheduleTime != nil {
-			earliestTime = cronJob.Status.LastScheduleTime.Time
-		} else {
-			earliestTime = cronJob.ObjectMeta.CreationTimestamp.Time
-		}
-		if cronJob.Spec.StartingDeadlineSeconds != nil {
-			// controller is not going to schedule anything below this point
-			schedulingDeadline := now.Add(-time.Second * time.Duration(*cronJob.Spec.StartingDeadlineSeconds))
-
-			if schedulingDeadline.After(earliestTime) {
-				earliestTime = schedulingDeadline
-			}
-		}
-		if earliestTime.After(now) {
-			return time.Time{}, sched.Next(now), nil
-		}
-
-		starts := 0
-		for t := sched.Next(earliestTime); !t.After(now); t = sched.Next(t) {
-			lastMissed = t
-			// An object might miss several starts. For example, if
-			// controller gets wedged on Friday at 5:01pm when everyone has
-			// gone home, and someone comes in on Tuesday AM and discovers
-			// the problem and restarts the controller, then all the hourly
-			// jobs, more than 80 of them for one hourly scheduledJob, should
-			// all start running with no further intervention (if the scheduledJob
-			// allows concurrency and late starts).
-			//
-			// However, if there is a bug somewhere, or incorrect clock
-			// on controller's server or apiservers (for setting creationTimestamp)
-			// then there could be so many missed start times (it could be off
-			// by decades or more), that it would eat up all the CPU and memory
-			// of this controller. In that case, we want to not try to list
-			// all the missed start times.
-			starts++
-			if starts > 100 {
-				// We can't get the most recent times so just return an empty slice
-				return time.Time{}, time.Time{}, fmt.Errorf("Too many missed start times (> 100). Set or decrease .spec.startingDeadlineSeconds or check clock skew.")
-			}
-		}
-		return lastMissed, sched.Next(now), nil
-	}
-	// +kubebuilder:docs-gen:collapse=getNextSchedule
```

Comment:
We will be able to make those changes here as well because this content is used un the tutorial
If the problem here is the complexity because the method is too long. Then, the best approach is to add on top of the fucn //nolint:complex <- The name of the linter to allow it to pass. 

Look the tutorial it will broke the docs.

Reply:
Addressed it, and used: `// nolint:gocyclo` for the Reconcile func. Thanks a lot for checking. Code is rebased on latest master.